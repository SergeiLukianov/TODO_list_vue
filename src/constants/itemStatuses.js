export const OPEN_STATUS = 'open'
export const DONE_STATUS = 'done'
