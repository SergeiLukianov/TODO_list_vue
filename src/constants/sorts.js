const options = {
  recordsAsc: 'Text ASC',
  recordsDesc: 'Text DESC',
  creationDateAsc: 'Date creation ASC',
  creationDateDesc: 'Date creation DESC',
  dueDateAsc: 'Due date ASC',
  dueDateDesc: 'Due date DESC',
};

export default {
  sortOptions: options,
  sortOptionsValues: options.keys(),
}
