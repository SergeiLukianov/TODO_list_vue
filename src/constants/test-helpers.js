const mockState = (store = {}) => JSON.parse(JSON.stringify(store));

const mockGetters = (getters = {}) => Object.keys(getters)
  .reduce((reducedGetters, name) => ({
    ...reducedGetters,
    [name]: jest.fn(),
  }), {});

const mockAction = () => jest.fn(() => Promise.resolve());
const mockActions = (actions = {}) => Object.keys(actions).reduce((mockedActions, name) => ({
  ...mockedActions,
  [name]: mockAction(),
}), {});

const mockMutations = (mutations = {}) => Object.keys(mutations).reduce((mockedMutations, name) => ({
  ...mockedMutations,
  [name]: jest.fn(),
}), {});

export function mockModules(modules) {
  return Object.keys(modules).reduce((mockedModules, name) => {
    let nextModule = modules[name];

    return {
      ...mockedModules,
      [name]: {
        namespaced: nextModule.namespaced,
        state: mockState(nextModule.state),
        getters: mockGetters(nextModule.getters),
        actions: mockActions(nextModule.actions),
        mutations: mockMutations(nextModule.mutations),
      }
    };
  }, {});
}

export function resetModules(modules) {
  return Object.keys(modules).reduce((resetedModules, nextModuleName) => ({
    ...resetedModules,
    [nextModuleName]: {
      ...modules[nextModuleName],
      state: JSON.parse(JSON.stringify(modules[nextModuleName].state || {})),
    },
  }), {});
}

function getNamespacedKeys(obj, moduleName, isNamespaced) {
  return Object.keys(obj).reduce((results, key) => ({
    ...results,
    [key]: (isNamespaced ? `${moduleName}/${key}` : key),
  }), {});
}

export function getActionTypes(moduleConfig) {
  const moduleName = Object.keys(moduleConfig)[0];
  const module = moduleConfig[moduleName];
  const isNamespaced = module.namespaced;

  return getNamespacedKeys(module.actions, moduleName, isNamespaced);
}

export function getGetterTypes(moduleConfig) {
  const moduleName = Object.keys(moduleConfig)[0];
  const module = moduleConfig[moduleName];
  const isNamespaced = module.namespaced;

  return getNamespacedKeys(module.getters, moduleName, isNamespaced);
}
