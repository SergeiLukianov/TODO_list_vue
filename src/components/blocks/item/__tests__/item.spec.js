import Item from '../item.vue'
import {createLocalVue, shallowMount} from '@vue/test-utils'
import * as STATUSES from '../../../../constants/itemStatuses'

const doneItem = {
  id: 1600433718398,
  status: STATUSES.DONE_STATUS,
  text: 'write test',
  dueDate: 1600433719398,
}

const openItem = {
  id: 1600433718398,
  status: STATUSES.OPEN_STATUS,
  text: 'write test',
}

const renderWrapper = (component, options = {}) => {
  if (!Boolean(options.localVue)) {
    options.localVue = createLocalVue()
  }

  return shallowMount(Item, {
    ...options,
  })
}

describe('Item', () => {

  it('renders open item', () => {
    const options = {
      propsData: {
        item: openItem,
      },
      data () {
        return {
          active: false,
          editMode: false,
        }
      },
    }
    expect(renderWrapper(Item, options).html()).toMatchSnapshot()
  });

  it('renders done item', () => {
    const options = {
      propsData: {
        item: doneItem,
      },
      data () {
        return {
          active: false,
          editMode: false,
        }
      },
    }
    expect(renderWrapper(Item, options).html()).toMatchSnapshot()
  });

  it('checkbox is unchecked for open item', () => {
    const options = {
      propsData: {
        item: openItem,
      }
    }
    const wrapper = renderWrapper(Item, options)

    expect(wrapper.find('.item input[type="checkbox"]').element.checked).toBe(false)
  });

  it('checkbox is checked for done item', () => {
    const options = {
      propsData: {
        item: doneItem,
      }
    }
    const wrapper = renderWrapper(Item, options)

    expect(wrapper.find('.item input[type="checkbox"]').element.checked).toBe(true)
  });

  it('created date in correct format', () => {
    const item = {
      id: new Date(2020, 9, 18, 19, 0).valueOf()
    }

    const options = {
      propsData: {
        item: item,
      }
    }
    const wrapper = renderWrapper(Item, options)

    expect(wrapper.vm.formattedCreationDate).toBe('7:00:00 PM')
  });
})
