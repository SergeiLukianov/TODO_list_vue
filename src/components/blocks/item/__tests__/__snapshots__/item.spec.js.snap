// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders done item 1`] = `
<div class="item">
    <div><input type="checkbox"></div>
    <div class="itemContent"><span>write test</span></div>
    <div class="dates-container">
        <div class="align-end center text-gray">
            3:55:18 PM
        </div>
        <div class="align-end center text-black">
            3:55:19 PM
        </div>
    </div>
    <div class="remove_item_btn">
        <!---->
    </div>
</div>
`;

exports[`Item renders open item 1`] = `
<div class="item">
    <div><input type="checkbox"></div>
    <div class="itemContent"><span>write test</span></div>
    <div class="dates-container">
        <div class="align-end center text-gray">
            3:55:18 PM
        </div>
        <!---->
    </div>
    <div class="remove_item_btn">
        <!---->
    </div>
</div>
`;
